### PR TITLE
Client Implementation for adding & removing a learning objects children

### PR DESCRIPTION
**Pull request by johngonz0**
Adds REMOVE_CHILD route in learning object routes for client.
Adds UPDATE_CHILDREN route in learning object routes for client.

Updates the setChildren function in learning-object.service

Updates all related functions parameters

Client changes for both service PRs - 
https://github.com/Cyber4All/clark-service/pull/141
https://github.com/Cyber4All/clark-service/pull/136

## Deleting a child:
**This works when you go in order from last child element to first,  but a bug was discovered when trying to go in another order**

https://github.com/Cyber4All/clark-client/assets/150100412/d74de4bf-f337-467b-bd7c-25f906e3adf0

**See bug here**
https://app.shortcut.com/clarkcan/story/31788/bug-removing-children-from-learning-object-in-builder

## Adding a child:
**Works great! if only deleting went this smooth... :(**

https://github.com/Cyber4All/clark-client/assets/150100412/ca1c5245-ed9e-45be-8d02-5f2e3e0cfdfe

### CODE REVIEW

**Review by mfranz1 — CHANGES_REQUESTED** [review]
The builder needs to be updated to actually remove the child. You can identify what route is currently being used by logging on to production and messing with a hierarchy in your author dashboard

**Review by JosueMolinaMorales — COMMENTED** [review]
Just to add to Mike's comment. I believe the route that updates the childrens array is `SET_CHILDREN` found [here](https://github.com/Cyber4All/clark-client/blob/clark-service-refactor/src/app/core/learning-object-module/learning-object/learning-object.routes.ts#L160)

I would just verify this by going into the builder and adding a child and then removing a child. Add console logs/look at network logs to help with this.

**Review by shanioffe — APPROVED** [review]

File: src/app/core/learning-object-module/learning-object/learning-object.routes.ts
Comment (mfranz1) [inline src/app/core/learning-object-module/learning-object/learning-object.routes.ts:1]: Please remove from LEGACY routes and move to LEARNING_OBJECT_ROUTES

File: src/app/onion/core/learning-object.service.ts [hunk 1]
```diff
@@ -470,17 +470,16 @@ export class LearningObjectService {
...
-    learningObjectName: string,
-    authorUsername: string,
+    learningObjectId: string,
     children: string[],
     remove: boolean,
   ): Promise<any> {
-    const route = LEGACY_USER_ROUTES.SET_CHILDREN(authorUsername, learningObjectName);
-
+    const removeRoute = LEGACY_USER_ROUTES.REMOVE_CHILD(learningObjectId);
+    const addRoute = LEGACY_USER_ROUTES.UPDATE_CHILDREN(learningObjectId);
```
Comment (mfranz1) [inline src/app/onion/core/learning-object.service.ts:478]: These will need to be updated once moved from LEGACY routes

**Review by mfranz1 — CHANGES_REQUESTED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]
just mikes comment

**Review by curlyLasagna — APPROVED** [review]

**Review by mfranz1 — APPROVED** [review]

**Review by pzalep1 — APPROVED** [review]

File: src/app/core/learning-object-module/learning-object/learning-object.routes.ts [hunk 2]
```diff
@@ -28,6 +28,23 @@ export const LEARNING_OBJECT_ROUTES = {
...
+     * Path to update the children of a learning object
+     * @param learningObjectId the id of the parent learning object
+     * @returns void
+     */
+    UPDATE_CHILDREN(learningObjectId: string) {
+        return `${environment.apiURL}/learning-objects/${learningObjectId}/children`;
+    },
+
+    /**
+     * Path to remove a children of a learning object
```
Comment (mfranz1) [inline src/app/core/learning-object-module/learning-object/learning-object.routes.ts:41]:
```suggestion
     * Path to remove a child of a learning object
```

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts [hunk 3]
```diff
@@ -145,20 +145,23 @@ export class ScaffoldComponent implements OnInit {
...
+  async deleteChild() {
     this.toggleConfirmationModal(false);
     // remove the child that was selected to be deleted
     this.children.splice(this.deleteIndex, 1);
 
     // set childrenIDs equal to the children array
     this.childrenIDs = [];
     this.children.forEach(kid => this.childrenIDs.push(kid._id));
-    this.store.setChildren(this.childrenIDs, true);
+    await this.store.setChildren(this.childrenIDs, true);
```
Comment (mfranz1) [inline src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts:156]: Huh, good find. Been like this for 7 years 😬

**Review by mfranz1 — APPROVED** [review]

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts (hunk 3 above)
Comment (johngonz0) [reply src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts:156]: yikes lol

**Review by curlyLasagna — APPROVED** [review]

**Review by pzalep1 — APPROVED** [review]

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts (hunk 3 above)
Comment (shanioffe) [reply src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts:156]: 😮

File: src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts [hunk 4]
```diff
@@ -145,20 +145,23 @@ export class ScaffoldComponent implements OnInit {
...
 
     // if deleted child was last child toggle off editContent because there is no longer content to edit
     if (this.children.length === 0) {
       this.editContent = false;
     }
+
+    // get the children again to get current childrens array
+    await this.store.getChildren();
```
Comment (shanioffe) [inline src/app/onion/learning-object-builder/components/scaffold/scaffold.component.ts:164]: You figured it out yayy 🥳

**Review by shanioffe — APPROVED** [review]

**Review by JosueMolinaMorales — APPROVED** [review]

**Pull request merged**